overlays/nodejs/serverless/function.ts

```diff
@@ -309,18 +313,20 @@ function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: stri
 // It is assumed that the tree was correctly construted such that dependencies are resolved to compatible versions in
 // the closest available match starting at the provided root and walking up to the head of the tree.
 function findDependency(root: Package, name: string) {
-    for(;root;root = root.parent) {
+    for(; root; root = root.parent) {
         for (var child of root.children) {
-            let childName = child.name;
-            // Note: `read-package-tree` returns incorrect `.name` properties for packages in an orgnaization - like
-            // `@types/express` or `@protobufjs/path`.  Compute the correct name from the `path` property instead.
-            // Match any name that ends with something that looks like `@foo/bar`.
-            const match = /\/\@[^\/]*\/[^\/]*$/.exec(child.path);
-            if (match) {
-                childName = match[0];
-            }
-            if(childName === name) {
-                return child;
+            if (name.indexOf("@") === -1) {
+                if (name === child.name) {
+                    return child;
+                }
+            } else {
+                // `read-package-tree` returns incorrect `.name` properties for packages in an orgnaization, like
+                // `@types/express` or `@protobufjs/path`.  Compute the correct name from the `path` property instead.
+                // We split the path name using the filesystem delimiter and look at the last two components.
+                const parts = child.path.split(filepath.sep);
```

Comment:
FWIW - I am not sure what if any normalization read-package-tree does on paths, and whether this change is more or less correct on Windows. Will be good to validate on Windows explicitly (though we have even less coverage at this layer on other OSs).